aten/src/ATen/native/TensorShape.cpp

```diff
@@ -3932,12 +3932,12 @@ at::Tensor as_strided_scatter_symint(const at::Tensor& self, const at::Tensor& s
 // If TLS is set appropriately (for wrapper-tensor keys like Functionalize or functorch transforms),
 // then we'll dispatch to one of their implementations, which will properly lift the tensor into a wrapper.
 at::Tensor lift(const at::Tensor& self) {
-    return self;
+    return self.view(self.sizes());
 }
 
 // See notes in native_functions.yaml
 at::Tensor lift_fresh(const at::Tensor& self) {
-    return self;
+    return self.view(self.sizes());
```

Comment:
cc @ezyang did this to appease tests. I think this is pretty harmless, since both of these kernels are CompositeExplicit so we'll never try to trace through the size calls. I'm not entirely clear on whether there was an existing silent correctness issue here with the autograd engine - my guess is that in practice, lift() calls only ever show up in a post-autograd graph, so we'll never try to autograd through them in the first place.

Reply:
Alright.